### PR TITLE
fix: return to ballotpopup

### PR DESCRIPTION
**Pull request by holloway**
* update the modal template route to take a param, rather than using request.path
* ensure every place that uses the ballot_icon template tag is added to the allow list of valid 'return to' route handlers.

### CODE REVIEW

**Review by jennifer-richards — DISMISSED** [review]

**Comment by holloway** [comment]
fyi @jennifer-richards: @rjsparks pointed out that the 'return to' param when coming from a modal doesn't make sense. Bootstrap modals are configured to scrape an modal template route (eg `/doc/draft-ietf-opsawg-ipfix-tcpo-v6eh/ballotpopup/998718/`) and then display that as a modal, so the 'return to' is the modal template route not the originating page from which the modal opened. This means there's a new layer of indirection when opening via modals, so the fix is to

* update the modal template route to take a param, rather than using `request.path`, and validate the param using the same logic as before.
* ensure every place that uses the ballot_icon template tag is added to the allow list of valid 'return to' route handlers. This is a bit like playing whack-a-mole, but I think I've found every usage.

**Comment by rjsparks** [comment]
> This is a bit like playing whack-a-mole, but I think I've found every usage.

**Review by jennifer-richards — CHANGES_REQUESTED** [review]
Looking good, but I think a valid return point was missed. There are a couple minor bits of cleanup I suggest.

File: ietf/doc/tests_ballot.py [hunk 1]
```diff
@@ -1457,7 +1457,7 @@ class ReturnToUrlTests(TestCase):
     def test_invalid_return_to_url(self):
         self.assertRaises(
             Exception,
-            lambda: parse_ballot_edit_return_point('/doc/', 'draft-ietf-opsawg-ipfix-tcpo-v6eh', '998718'),
+            lambda: parse_ballot_edit_return_point('/', 'draft-ietf-opsawg-ipfix-tcpo-v6eh', '998718'),
```
[the remark this answers is not in the document]
Comment (holloway) [reply ietf/doc/tests_ballot.py]: Done

File: ietf/doc/templatetags/ballot_icon.py [hunk 2]
```diff
@@ -96,9 +96,14 @@ def sort_key(t):
...
 
+    request = context.get("request")
+    ballot_edit_return_point_param = f"?ballot_edit_return_point={request.path}"
+
     right_click_string = ''
     if has_role(user, "Area Director"):
-        right_click_string = 'oncontextmenu="window.location.href=\'%s\';return false;"' %  urlreverse('ietf.doc.views_ballot.edit_position', kwargs=dict(name=doc.name, ballot_id=ballot.pk))
+        right_click_string = 'oncontextmenu="window.location.href=\'%s%s\';return false;"' % (
+            urlreverse('ietf.doc.views_ballot.edit_position', kwargs=dict(name=doc.name, ballot_id=ballot.pk)),
+            ballot_edit_return_point_param)
```
[the remark this answers is not in the document]
Comment (holloway) [reply ietf/doc/templatetags/ballot_icon.py:106]: Good idea. Done

**Review by jennifer-richards — APPROVED** [review]
Lgtm. I didn't have a chance to double check that all the places `search_result_row.html` is included are in the allowed view list.

**Comment by holloway** [comment]
@jennifer-richards fyi here's my notes on tracing back from `search_result_row.html`

```
search_result_row.html
	search_results.html
		view_list.html
			ietf/community/views.view_list
		drafts_for_ad.html
			ietf/doc/views_search.docs_for_ad
		drafts_in_last_call.html
			ietf/doc/views_search.drafts_in_last_call
		recent_drafts.html
			ietf/doc/views_search.recent_drafts
		search/search.html
			ietf/doc/views_search.search
		group_documents.html
			ietf/group/views.group_documents
		stream_documents.html
			ietf/group/views.stream_documents
	agenda_documents.html
		ietf/iesg/views.agenda_documents
```

(those views are in the allowed list)

**Review by rjsparks — APPROVED** [review]

**Pull request merged**